test/torch/test_hook.py

```diff
@@ -293,23 +293,68 @@ def initHidden(self):
         param.data.add_(-learning_rate, param.grad.data)
 
 
-def test_remote_gradient_clipping(workers):
-    # Vanishing gradient test
+def test_local_remote_gradient_clipping(workers):
+    """
+    Real test case of gradient clipping for the remote and
+    local parameters of an RNN
+    """
     alice = workers["alice"]
-    vanishing_tensor_test = torch.Tensor([-9.8367e23])
-    remote_vanishing_tensor = vanishing_tensor_test.send(alice)
-    vanishing_remote_tensor_clipped = torch.nn.utils.clip_grad(remote_vanishing_tensor, 2)
-    # Has the remote gradient indeed increased?
-    greater_tensor_check = (vanishing_remote_tensor_clipped > remote_vanishing_tensor).copy().get()
-    one_tensors = torch.ones([1], dtype=torch.uint8)
-    assert torch.eq(greater_tensor_check, one_tensors)
-
-
-def test_local_gradient_clipping():
-    # Vanishing gradient test
```

Comment:
is there a reason this was removed? it isn't inherently a bad test

Reply:
Because in a real scenario, this method is supposed to be applied to a generator of parameters of a model and not to one simple tensor. The new tests checks pretty much the same things as the previous anyway.

Reply:
There is a difference between unit and integration tests. The deleted test seems a bit closer to a unit test but it seems you replaced it with an integration test